### ISSUE TITLE
Reindex fails due to "mapper_parsing_exception"

### ISSUE THREAD

**Issue by leowmjw** [Sinar/popit_ng#130]
Reindex fails for current Popit Production dataset.

The data set is from the production data fixture -->  http://data.sinarproject.org/dataset/popit-ng-production-fixture

**(virtualenv) vagrant@precise64:/vagrant$ ./manage.py reindex**
```
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
...
 File "/home/vagrant/virtualenv/local/lib/python2.7/site-packages/elasticsearch/connection/base.py", line 105, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
elasticsearch.exceptions.RequestError: TransportError(400, u'mapper_parsing_exception', u'failed to parse')
```

Relevant log
```
==> log/elasticsearch.log <==
HEAD http://elasticsearch.service.consul:9200/popit [status:200 request:0.005s]
> None
< 
curl -XHEAD 'http://localhost:9200/popit?pretty' -d ''
#[200] (0.005s)
#
GET http://elasticsearch.service.consul:9200/popit/persons/_search?q=id%3A53659684f1eab6270da6c8fc+AND+language_code%3Aen [status:200 request:0.008s]
> None
< {"took":6,"timed_out":false,"_shards":{"total":5,"successful":5,"failed":0},"hits":{"total":0,"max_score":null,"hits":[]}}
curl -XGET 'http://localhost:9200/popit/persons/_search?pretty&q=id%3A53659684f1eab6270da6c8fc+AND+language_code%3Aen' -d ''
#[200] (0.008s)
#{
#  "_shards": {
#    "failed": 0,
#    "successful": 5,
#    "total": 5
#  },
#  "hits": {
#    "hits": [],
#    "max_score": null,
#    "total": 0
#  },
#  "timed_out": false,
#  "took": 6
#}
POST /popit/persons [status:400 request:0.007s]
> {"id": "53659684f1eab6270da6c8fc", "other_names": [], "identifiers": [], "memberships": [{"id": "536597c919ee29270d8a9efc", "person_id": "53659684f1eab6270da6c8fc", "organization_id": null, "member_id": null, "on_behalf_of_id": null, "area_id": null, "post_id": "536597b019ee29270d8a9efb", "contact_details": [], "links": [], "start_date": "2013", "end_date": "2016", "created_at": "2015-12-21T02:02:18.293043Z", "updated_at": "2015-12-21T02:02:18.293090Z", "member": null, "on_behalf_of": null, "label": null, "role": null, "language_code": null}, {"id": "55a5fae7c49fbcbe51072688", "person_id": "53659684f1eab6270da6c8fc", "organization_id": "55a5e6b54ae9a27d44ef6cf0", "member_id": null, "on_behalf_of_id": null, "area_id": null, "post_id": null, "contact_details": [], "links": [], "start_date": "2013-12-20", "end_date": null, "created_at": "2015-12-21T06:50:17.391972Z", "updated_at": "2015-12-21T06:50:17.392036Z", "member": null, "on_behalf_of": null, "label": null, "role": null, "language_code": null}], "links": [], "contact_details": [], "birth_date": null, "death_date": null, "email": "", "image": "", "created_at": "2015-12-16T06:04:58.152925Z", "updated_at": "2016-02-11T09:08:32.019204Z", "name": "Azih Muda", "family_name": "Azih Muda", "given_name": null, "additional_name": null, "honorific_prefix": null, "honorific_suffix": null, "patronymic_name": null, "sort_name": null, "gender": "Male", "summary": "President of CUEPECS", "biography": "", "national_identity": null, "language_code": "en"}
```

**Comment by sweemeng** [comment]
I think currently automatically try to parse *_date field with proper datetime. I have issue in development in environment where the data with *_date field cannot be indexed due to formatting reason

**Comment by sweemeng** [comment]
What I did is I have to fix the fixtures. I think I would commit a patch that convert YYYY to YYYY0101 00:00:00

**Comment by sweemeng** [comment]
@leowmjw please verify, fix is in #148
